#!/usr/bin/env python
PACKAGE = "flypulator_mavros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("drone_armed",     bool_t, 0, "Arming state of drone ", False) 

gen.add("upper_limit",     double_t, 0, "upper limit of rpm", 600, 50, 6000) 
gen.add("motor1_speed",     double_t, 0, "Speed of motor 1 in range ", 0, 0, 6000) 
gen.add("motor2_speed",     double_t, 0, "Speed of motor 2 in range ", 0, 0, 6000) 
gen.add("motor3_speed",     double_t, 0, "Speed of motor 3 in range ", 0, 0, 6000) 
gen.add("motor4_speed",     double_t, 0, "Speed of motor 4 in range ", 0, 0, 6000) 
gen.add("motor5_speed",     double_t, 0, "Speed of motor 5 in range ", 0, 0, 6000) 
gen.add("motor6_speed",     double_t, 0, "Speed of motor 6 in range ", 0, 0, 6000) 

exit(gen.generate(PACKAGE, "flypulator_mavros", "offb_parameter"))
